src/Aspire.Dashboard/Configuration/DashboardOptions.cs

```diff
@@ -305,3 +306,45 @@ internal bool TryParseOptions([NotNullWhen(false)] out IEnumerable<string>? erro
         return messages is null;
     }
 }
+
+public sealed class DebugSessionOptions
+{
+    private X509Certificate2? _serverCertificate;
+
+    public string? Address { get; set; }
+    public string? Token { get; set; }
+    public string? ServerCertificate { get; set; }
+    public bool? TelemetryOptOut { get; set; }
+
+    public X509Certificate2? GetServerCertificate() => _serverCertificate;
+
+    internal bool TryParseOptions([NotNullWhen(false)] out string? errorMessage)
+    {
+        if (!string.IsNullOrEmpty(ServerCertificate))
+        {
+            byte[] data;
+            try
+            {
+                data = Convert.FromBase64String(ServerCertificate);
+            }
+            catch (Exception ex)
+            {
+                errorMessage = $"Error converting server certificate payload from base64 to bytes: {ex.Message}";
+                return false;
+            }
+
+            try
+            {
+                _serverCertificate = new X509Certificate2(data);
+            }
+            catch (Exception ex)
+            {
+                errorMessage = $"Error reading server certificate as X509Certificate2: {ex.Message}";
+                return false;
+            }
+        }
+
+        errorMessage = null;
```

Comment:
Are there any expectations for other data (`Address` and `Token`)?

Reply:
In particular, should we verify that the address is always local?

Reply:
Is there extra validation in DCP? For example, do you double check the address is local?

We can apply the same validation here.

Reply:
DCP does not allow overriding the address, we always use `localhost`; the thing that gets specified by the env vars is the port https://github.com/dotnet/aspire/blob/main/docs/specs/IDE-execution.md#enabling-ide-execution. The port is not validated directly, but we do validate the resulting connection URL.

Token is used as-is.

Reply:
Updated